File: 06.bean-conditional/src/main/java/study/spring/beanconditional/v1/TruePrinterConfiguration.java
package study.spring.beanconditional.v1;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import study.spring.beanconditional.v1.model.BooleanPrinterV1;
import study.spring.beanconditional.v1.model.TruePrinterV1;

@Configuration
@ConditionalOnProperty(value = "application.printer.boolean-printer.v1", havingValue = "true")
public class TruePrinterConfiguration {

  @Bean
  public BooleanPrinterV1 booleanPrinter() {
    return new TruePrinterV1();
  }
}

File: bean-conditional/src/main/kotlin/study/spring/beanconditional/TruePrinterConfiguration.kt
package study.spring.beanconditional

import org.springframework.context.annotation.*
import org.springframework.core.type.AnnotatedTypeMetadata

@Configuration
@Conditional(TruePrinterCondition::class)
class TruePrinterConfiguration {

    @Bean
    fun booleanPrinter(): BooleanPrinter = TruePrinter()
}

class TruePrinterCondition: Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        return true
    }

}